lib/jackalope/transient_work_list.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Jackalope.TransientWorkList do
+  @moduledoc false
+
+  alias Jackalope.WorkList
+
+  @behaviour WorkList
+  require Logger
+  @default_max_size 100
+
+  defstruct items: [],
+            pending: %{},
+            max_size: @default_max_size,
+            expiration_fn: nil
+
+  @type t() :: %__MODULE__{items: list(), max_size: non_neg_integer()}
+
+  @impl true
```

Comment:
For consistency with the persistent work_list:

```suggestion
  @impl WorkList
```